R/estimate_infections.R

```diff
@@ -237,8 +207,56 @@ estimate_infections <- function(reported_cases,
                                 future = FALSE, 
                                 max_execution_time = Inf, 
                                 return_fit = FALSE,
+                                id = "estimate_infections",
                                 verbose = FALSE){
   
+  # Parse Stan Arguments to set algorithm and backend
+  backend <- match.arg(stan_configuration[["backend"]], backends_available())
+  method <- match.arg(stan_configuration[["algorithm"]], algoriths_available())
+  
+  method <- ifelse(method!="sampling", "approximate", method)
+  
+  # If they are null, a null will be passed per the defaults
+  
+  cache_model <- ifelse(is.null(stan_configuration[["cache_model"]]),TRUE, stan_configuration[["cache_model"]])
+  model <- stan_configuration[["model"]]
+  fit <- stan_configuration[["fit"]]
+  
+  # Check Input of Model
+  # Collapse multiple fits if they are passed
+  if(!is.null(fit)){
+    if(class(fit)=="list"){
+      if(any(lapply(fit, class)!="stanfit")){
+        stop("The fit you have passed is not a valid `stanfit` object")
+      } else{
+        # If a list was passed, combine to a single rstan fit object
+        fit <- rstan::sflist2stanfit(fit)
+      }
+    } else {
+      if(class(fit)!="stanfit"){
+        stop("The fit you have passed is not a valid `stanfit` object")
+      }
+    }
+  }
+  
+  
+  
+  # Nullify non-standard arguments now for downstream compliance
+  stan_args$backend <- NULL
+  stan_args$method <- NULL
+  stan_args$model <- NULL
+  stan_args$fit <- NULL
+  stan_args$cache_model <- NULL
```

Comment:
Instead of nullifying the non-standard arguments, we probably want to generalize `make_cmdstan_list()` as `standardize_stan_args()` with a `backend` argument, and then call it before fitting the model, just like in `CmdStan` backend. It can also replace `create_stan_args()`. So, we end up with a single well-document function for handling and formatting Stan arguments for all backends and algorithms.

Reply:
I like this approach. In latest push (b5524ec) I added `create_standardised_stan_args` that parses the `stan_args` list that is passed into the `estimate_infections`. This opens the possibility for additional re-factoring down the line as needed as well.

Reply:
Looking great! I think that we can merge `create_standardised_stan_args` (which has a long name) into `create_stan_args`, and, if needed, add a logical argument `standard` to switch between the two functionalities.